Library/Homebrew/extend/os/linux/diagnostic.rb

```diff
@@ -154,6 +155,22 @@ def check_linuxbrew_bottle_domain
           EOS
         end
 
+        def check_for_symlinked_home
+          return unless File.symlink?("/home")
+
+          <<~EOS
+            Your /home directory is a symlink.
+            This is known to cause issues with formula linking, particularly when installing
+            multiple formulae that create symlinks in shared directories.
+
+            We recommend using systems with standard directory structures.
+            If you encounter linking issues, you may need to manually create conflicting
+            directories or use `brew link --overwrite` as a workaround.
```

Comment:
Clarify in this text that it may be a standard directory structure in the distribution being used e.g. Fedora Silverblue and we'd welcome a PR to fix this functionality but there's known issues as-is